lib/cucumber/formatter/json.rb

```diff
@@ -11,13 +11,13 @@ class Json
       include Io
 
       def initialize(config)
-        config.on_event :before_test_case, &method(:on_before_test_case)
-        config.on_event :after_test_case, &method(:on_after_test_case)
-        config.on_event :before_test_step, &method(:on_before_test_step)
-        config.on_event :after_test_step, &method(:on_after_test_step)
-        config.on_event :finished_testing, &method(:on_finished_testing)
         @io = ensure_io(config.out_stream)
         @feature_hashes = []
+        config.on_event :test_case_starting, &method(:on_before_test_case)
```

Comment:
I think it would be better to name the methods call on an event with the pattern: `on_<event_name>`